### PR TITLE
Publicfunction api

### PR DESCRIPTION
**Pull request by kKahina**
Build API for totally public functions

### CODE REVIEW

File: src/boilerplate/orchestration/javascript/raw/toOrchestration.ts [hunk 1]
```diff
@@ -873,6 +873,33 @@ export const OrchestrationCodeBoilerPlate: any = (node: any) => {
...
+          \nconst key = config.web3.key;
+          \nconst signed = await web3.eth.accounts.signTransaction(txParams, key);
+          \nconst sendTxn = await web3.eth.sendSignedTransaction(signed.rawTransaction);
+          \nlet tx = await instance.getPastEvents("NewLeaves");
+          \ntx = tx[0];
+          \nif (!tx) {
+            throw new Error('Tx failed - the commitment was not accepted on-chain, or the contract is not deployed.');
+          }`
+        ]
+      };*/
```
Comment (SwatiEY) [inline src/boilerplate/orchestration/javascript/raw/toOrchestration.ts:902]: Do we need this commented code.

File: src/transformers/visitors/toOrchestrationVisitor.ts [hunk 2]
```diff
@@ -409,6 +409,59 @@ const visitor = {
...
+        newNode.sendTransactionNode = buildNode('SendTransaction', {
+          functionName: node.fileName,
+          contractName,
+        });
+        console.log('Node in vistor:', node);
+        node.parameters.parameters.forEach(para =>{
+        newNode.sendTransactionNode.publicInputs.push(para.name);
+})
+
+/newNode.nodes.push(newNode.sendTransactionNode);
```
Comment (SwatiEY) [inline src/transformers/visitors/toOrchestrationVisitor.ts:435]: This node in the Public function AST generates the same code as we need for functions with secret states, you need to create a node separate for public functions.

Comment (kKahina) [reply src/transformers/visitors/toOrchestrationVisitor.ts:435]: I don't think there is a need to create a new one, juste the "nodeType": "FunctionDefinitionPublic", that should be new

File: src/boilerplate/common/services/generic-apiPublic_services.mjs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    await startEventFilter('CONTRACT_NAME');
+    const FUNCTION_SIG;
+    const { tx , _RESPONSE_} = await FUNCTION_NAME(FUNCTION_SIG);
+    // prints the tx
+    console.log(tx);
+    res.send({tx, _RESPONSE_});
+    // reassigns leafIndex to the index of the first commitment added by this function
+    if (tx.event) {
+      // prints the new leaves (commitments) added by this function call
+      console.log(`Merkle tree event returnValues:`);
```
Comment (lydiagarms) [inline src/boilerplate/common/services/generic-apiPublic_services.mjs:32]: We probably don't need this console log if the Merkle tree doesn't change?

File: src/boilerplate/orchestration/javascript/nodes/boilerplate-generator.ts [hunk 4]
```diff
@@ -265,6 +265,21 @@ export function buildBoilerplateNode(nodeType: string, fields: any = {}): any {
         publicInputs,
       };
     }
+    /*case 'FunctionDefinitionPublic': {
```
Comment (lydiagarms) [inline src/boilerplate/orchestration/javascript/nodes/boilerplate-generator.ts:268]: We can probably delete the commented code?

File: src/codeGenerators/orchestration/files/toOrchestration.ts [hunk 5]
```diff
@@ -132,16 +132,37 @@ const prepareIntegrationApiServices = (node: any) => {
...
   relevantFunctions.forEach((fn: any) => {
-  let fnboilerplate = genericApiServiceFile.postStatements()
+  let fnboilerplate = fn.nodeType === 'IntegrationApiServiceFunction'?
+  genericApiServiceFile.postStatements()[0]
+    .replace(/CONTRACT_NAME/g, node.contractName)
+    .replace(/FUNCTION_NAME/g, fn.name): genericApiServiceFile.postStatements()[1]
     .replace(/CONTRACT_NAME/g, node.contractName)
-    .replace(/FUNCTION_NAME/g, fn.name);
+    .replace(/FUNCTION_NAME/g, fn.name) ;
+    console.log("fnboilerplate :", fnboilerplate);
```
Comment (lydiagarms) [inline src/codeGenerators/orchestration/files/toOrchestration.ts:165]: Can we remove this console log?

File: src/transformers/visitors/common.ts [hunk 6]
```diff
@@ -42,6 +42,10 @@ export const initialiseOrchestrationBoilerplateNodes = (fnIndicator: FunctionDef
     functionName: node.fileName,
     contractName,
   });
+ // newNodes.functionPublicNode = buildNode('FunctionDefinitionPublic', {
```
Comment (lydiagarms) [inline src/transformers/visitors/common.ts:45]: Can we remove these comments?

File: src/transformers/visitors/toOrchestrationVisitor.ts [hunk 7]
```diff
@@ -409,6 +409,59 @@ const visitor = {
...
+
+       // Ajout des entrées publiques
```
Comment (lydiagarms) [inline src/transformers/visitors/toOrchestrationVisitor.ts:437]: Can you leave the comments in English.

**Pull request closed without merging**